packages/react-components/src/components/ChatMessage/MyMessageComponents/ChatMyMessageComponent.tsx

```diff
@@ -113,26 +121,35 @@ export const ChatMyMessageComponent = (props: ChatMyMessageComponentProps): JSX.
       onDeleteMessage(clientMessageId);
     }
   }, [onDeleteMessage, message.messageId, message.messageType, clientMessageId]);
+
   const onResendClick = useCallback(() => {
     onDeleteMessage && clientMessageId && onDeleteMessage(clientMessageId);
-    onSendMessage && onSendMessage(content !== undefined ? content : '');
-  }, [clientMessageId, content, onSendMessage, onDeleteMessage]);
+    onSendMessage &&
+      onSendMessage(
+        content !== undefined ? content : '',
+        /* @conditional-compile-remove(attachment-upload) */
+        {
+          attachmentMetadata: (message as ChatMessage).attachments
+        }
+      );
+  }, [onDeleteMessage, clientMessageId, onSendMessage, content, message]);
 
   if (isEditing && message.messageType === 'chat') {
     return (
       <ChatMessageComponentAsEditBoxPicker
         message={message}
         strings={props.strings}
-        onSubmit={async (text, metadata, options) => {
+        onSubmit={async (text, /* @conditional-compile-remove(attachment-upload) */ attachmentMetadata) => {
```

Comment:
It might be a good idea to move this functionality to a separate const that uses `useCallback` hook